src/test/regress/upgrade/custom_citus.py

```diff
@@ -0,0 +1,151 @@
+#!/usr/bin/env python3
+
+"""custom_citus
+Usage:
+    custom_citus --bindir=<bindir> --pgxsdir=<pgxsdir> --parallel=<parallel>
+
+Options:
+    --bindir=<bindir>              The PostgreSQL executable directory(ex: '~/.pgenv/pgsql-11.3/bin')
+    --pgxsdir=<pgxsdir>           	       Path to the PGXS directory(ex: ~/.pgenv/src/postgresql-11.3)
+    --parallel=<parallel>           how many configs to run in parallel
+"""
+
+import upgrade_common as common
+import threading
+import atexit
+from docopt import docopt
+import os, shutil
+import time
+import sys
+
+from config import (
+    CitusDefaultClusterConfig, CitusSingleNodeClusterConfig,
+    CitusSingleNodeSingleShardClusterConfig,
+    CitusSingleShardClusterConfig,
+    CitusNonMxClusterConfig,
+    CitusManyShardsClusterConfig,
+    CitusSingleNodeSingleSharedPoolSizeClusterConfig,
+    CitusSingleNodeSingleConnectionClusterConfig,
+    CitusSingleWorkerClusterConfig,
+    CitusShardReplicationFactorClusterConfig,
+    CitusNoLocalExecutionClusterConfig,
+    CitusComplexClusterConfig,
+    CitusSuperUserDefaultClusterConfig,
+    CITUS_CUSTOM_TEST_DIR,
+    CUSTOM_TEST_NAMES,
+    COORDINATOR_NAME,
+    CUSTOM_CREATE_SCHEDULE,
+    CUSTOM_SQL_SCHEDULE,
+    REGULAR_USER_NAME
+)
+
+testResults = {}
+failCount = 0
+
+def run_for_config(config, name):
```

Comment:
I realized that sometimes the threads are kept waiting forever(in case of failure), there is either something not finishing in this method or there is a way to exit from this method that doesn't signal "finished" to the thread. Any idea?

Reply:
I think that is probably related to  setting `shared_pool_size = 1`